content/actions/using-workflows/workflow-syntax-for-github-actions.md

```diff
@@ -342,6 +342,34 @@ steps:
     uses: actions/heroku@1.0.0
 ```
 
+### Example: Using secrets
+
+Secrets cannot be directly referenced in `if:` conditionals. Instead, consider setting secrets as job-level environment variables, then referencing the environment variables to conditionally run steps in the job.
+
+The example assumes there is a secret named `SECRET_IS_SET`.
+
+{% raw %}
+```yaml
+name: Run a step if a secret has been set
+on: push
+jobs:
+  my-jobname:
+    runs-on: ubuntu-latest
+    env:
+      SECRET_IS_SET: ${{ secrets.SECRET_IS_SET }}
+      SECRET_IS_NOT_SET: ${{ secrets.SECRET_IS_NOT_SET }}
+    steps:
+      - if: ${{ env.SECRET_IS_SET }}
```

Comment:
This will be truthy, but I wonder if an explicit comparison to an empty string would really drive the point home?

Reply:
Good suggestion @ethomson, thank you. -> aa53817

Rather than referencing two secrets (`${{ secrets.SECRET_IS_SET }}` and `${{ secrets.SECRET_IS_NOT_SET }}`), I rewrote this section to reference a single secret (`${{ secrets.SECRET_IS_SET }}`), and updated the `if:` conditionals to compare with empty strings as suggested. Hopefully this is a little clearer.

Reply:
Updated again to use same environment variable and secret names as the cross-referenced "Encrypted secrets" page (e0b2cfb).